models/middlewares/logChange.js

```diff
@@ -0,0 +1,45 @@
+import logger from "@config/logger";
+import { Changelog } from "@models/index";
+
+export default async function logChange({ userId, collection, changesBefore, changesAfter }) {
+  let diff;
+
+  const collectionName = collection;
+  const docId = changesAfter._id || changesBefore._id;
+
+  if (changesBefore && changesAfter) {
+    diff = getChangesDiff(changesBefore, changesAfter);
+  } else {
+    diff = changesAfter;
+  }
+
+  const change = {
+    userId,
+    docId,
+    collectionName,
+    changesBefore,
+    changesAfter,
+    diff,
+  };
+
+  if (changesBefore !== changesAfter && JSON.stringify(diff) !== "{}") {
+    await Changelog.create({ ...change });
+    logger.info(change);
```

Comment:
I think this logs too much info, it creates pages each time. Maybe something like this...

```suggestion
    logger.info(`Changelog added for: "${collection}"`);
```